src/Type/Addon/PointType.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Graphpinator\Type\Addon;
+
+final class PointType extends \Graphpinator\Type\Type
+{
+    protected const NAME = 'Point';
+    protected const DESCRIPTION = 'Point type - float which contains points x and y.';
+
+    protected function getFieldDefinition() : \Graphpinator\Field\ResolvableFieldSet
+    {
+        return new \Graphpinator\Field\ResolvableFieldSet([
+            new \Graphpinator\Field\ResolvableField(
+                'x',
+                \Graphpinator\Container\Container::Float()->notNull(),
+                static function(\stdClass $point) {
+                    return $point->x;
+                },
+            ),
+            new \Graphpinator\Field\ResolvableField(
+                'y',
+                \Graphpinator\Container\Container::Float()->notNull(),
+                static function(\stdClass $point) {
+                    return $point->y;
+                },
+            ),
+        ]);
+    }
+
+    protected function validateNonNullValue($rawValue) : bool
+    {
+        return $rawValue instanceof \stdClass
+            && \property_exists($rawValue, 'x')
+            && \property_exists($rawValue, 'y')
+            && \is_float($rawValue->x)
```

Comment:
Either remove the is_float conditions or also add them to GpsType.

Also check Rgb/HslType and make sure it is uniformed across all types.